presto-main/src/main/java/com/facebook/presto/metadata/FunctionAndTypeManager.java

```diff
@@ -199,18 +191,16 @@ public Type getType(TypeSignature signature)
             if (type.isPresent()) {
                 return type.get();
             }
-            try {
-                return parametricTypeCache.getUnchecked(signature);
-            }
-            catch (UncheckedExecutionException e) {
-                throwIfUnchecked(e.getCause());
-                throw new RuntimeException(e.getCause());
-            }
         }
 
         Optional<FunctionNamespaceManager<?>> functionNamespaceManager = getServingFunctionNamespaceManager(signature.getTypeSignatureBase());
         checkArgument(functionNamespaceManager.isPresent(), "Cannot find function namespace for type '%s'", signature.getBase());
-        return instantiateParametricType(signature, functionNamespaceManager.get());
+        Optional<UserDefinedType> userDefinedType = functionNamespaceManager.get().getUserDefinedType(signature.getTypeSignatureBase().getQualifiedObjectName());
+        if (!userDefinedType.isPresent()) {
+            throw new IllegalArgumentException("Unknown type " + signature);
+        }
+        checkArgument(userDefinedType.get().getPhysicalTypeSignature().getTypeSignatureBase().isStandardType(), "UserDefinedType must be based on static types.");
+        return getType(userDefinedType.get().getPhysicalTypeSignature());
```

Comment:
What is a "static type"?

Reply:
Built in types / types that's defined in code thus cannot be changed. Do you have a better name for them?